0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md

```diff
@@ -33,31 +33,37 @@ Any SPSP receiver will run an SPSP server and expose an HTTPS endpoint called th
+2. The SPSP Endpoint responds with the receiver info, including the receiver's ILP address and the shared secret to be used in PSK2. It MAY respond with a balance associated with this SPSP receiver, i.e. in the case of an invoice.
 3. The sender constructs an ILP payment using the receiver's ILP address.
-4. The sender uses PSK to generate the payment condition and format additional data intended for the reciever to be sent with the payment.
-5. The sender prepares a local transfer to a connector with the condition and ILP packet attached.
-6. The receiver's PSK module registers the incoming transfer, parses and validates the ILP packet and the incoming transfer.
-7. The receiver MAY submit the incoming payment to an external system for review to ensure that the funds are wanted.
-8. If the payment is expected, the receiver's PSK module submits the condition fulfillment to the ledger to execute the transfer. If not, the PSK module rejects the incoming transfer.
-9. If the receiver executed the transfer, the sender's SPSP client receives a notification from its ILP module that the transfer has been executed, including the condition fulfillment from the receiver, and notifies the sender that the payment is completed. If the receiver rejected the transfer, the sender's SPSP client receives a notification with the receiver-specified error message detailing why the payment was rejected.
+4. The sender begins sending the payment.
+  1. The sender uses PSK2 to generate the payment chunk and format additional dthese stepsata intended for the reciever to be sent with the payment.
+  2. The sender sends a prepare packet to a connector with the condition and ILP address produced by PSK2.
+  3. The receiver's PSK2 module registers the incoming packet, parses, and validates the ILP packet.
+  4. The receiver MAY submit the incoming packet to an external system for review to ensure that the funds are wanted.
+  5. If the payment is expected, the receiver's PSK2 module derives a fulfillment packet from the prepare packet. If not, the PSK2 module replies with a reject packet.
+  6. The receiver MAY update a balance related to this SPSP receiver, i.e. in the case of an invoice.
+  7. The fulfillment or rejection packet comes back to the sender.
+  8. The sender MAY send further payment chunks with PSK2 (repeating these steps).
 
 ## Specification
 
-The SPSP endpoint is a URI used by the sender to query information about the receiver and set up payments. The SPSP endpoint URI MAY contain query string parameters. The sender SHOULD treat the URI as opaque.
+The SPSP endpoint is a URI used by the sender to query information about the receiver (which may be an invoice) and set up payments. The SPSP endpoint URI MUST NOT contain query string parameters. The sender SHOULD treat the URI as opaque. There are several supported ways to refer to an SPSP endpoint:
```

Comment:
Why can't the spsp uri contain query parameters?

Reply:
It seems like a strange way to use the identifier, both visually (keeping special characters down is nice for readability) and also lets people do things like `https://wallet.example.com/?receiver=$bob.example.com` without worrying about escaping. It also creates some more complexity in how we resolve the identifier by adding more moving parts. Does `$example.com?foo=bar` resolve to `https://example.com/.well-known/pay?foo=bar` or `https://example.com/?foo=bar`? It would also preclude us from adding any query parameters as part of the resolution protocol, if we did that sometime in the future.

I don't think any of those are super strong reasons, but I also don't think the benefit is worth it.